godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -252,8 +316,43 @@ namespace godel_surface_detection
     }
 
 
+    template <typename T>
+    static boost::shared_ptr<pcl::PointCloud<T>> downsampleCloud(const pcl::PointCloud<T>& cloud,
+                                                                 std::default_random_engine& random_engine,
+                                                                 int one_in)
+    {
```

Comment:
Could use a little documentation here. Even knowing what the function is supposed to accomplish, it took me a little while to understand the "one_in" argument. A brief would make the other comments much more readable/understandable.